### PR TITLE
feat(issue-platform): Integrate policy layer into issue search

### PR DESCRIPTION
**Pull request by wedamija**
This adds in the policy layer to the issue search, so that issue types will only appear in search when they're been released.

We'll need to change how issue search works once performance issues are moved over. Issue types that use the same dataset should be grouped into the same query, but at the moment we'll make a separate query for each category.

### CODE REVIEW

**Comment by wedamija** [comment]
Still need to write tests, just wanted to get something up.

**Comment by barkbarkimashark** [comment]
This looks reasonable to me.

**Review by roggenkemper — APPROVED** [review]

File: tests/snuba/search/test_backend.py [hunk 1]
```diff
@@ -2362,6 +2363,44 @@ def test_generic_query(self):
...
+                    },
+                )
+
+            results = self.make_query(search_filter_query="issue.category:performance my_tag:2")
+            assert list(results) == []
+            with self.feature(
+                [
+                    "organizations:issue-platform",
+                    group_type.build_visible_feature_name(),
+                    "organizations:performance-issues-search",
```
Comment (barkbarkimashark) [inline tests/snuba/search/test_backend.py:2398]: Damn, it looks like this feature flag might just be stuck here for S4S:
https://github.com/getsentry/getsentry/blob/e93b93aa619f050a9488a3244ee65fd41aa77c33/getsentry/conf/settings/singletenant.py#L541

**Review by barkbarkimashark — APPROVED** [review]

File: tests/snuba/search/test_backend.py (hunk 1 above)
Comment (wedamija) [reply tests/snuba/search/test_backend.py:2398]: Yeah, we'll hopefully be able to get rid of that soon

**Pull request merged**